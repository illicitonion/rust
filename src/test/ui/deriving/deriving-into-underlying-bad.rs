// Test that IntoUnderlying cannot be derived other than for enums with an int repr.

#![allow(unused)]

#[derive(IntoUnderlying)]
//~^ ERROR `IntoUnderlying` can only be derived for enums with an explicit integer representation [FIXME]
struct Struct {}

#[derive(IntoUnderlying)]
//~^ ERROR `IntoUnderlying` can only be derived for enums with an explicit integer representation [FIXME]
#[repr(C)]
enum NumberC {
    Zero,
    One,
}

#[derive(IntoUnderlying)]
//~^ ERROR `IntoUnderlying` can only be derived for enums with an explicit integer representation [FIXME]
enum NumberNoRepr {
    Zero,
    One,
}

fn main() {}
