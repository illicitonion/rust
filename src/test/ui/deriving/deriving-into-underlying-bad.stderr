error[FIXME]: `IntoUnderlying` can only be derived for enums with an explicit integer representation
  --> $DIR/deriving-into-underlying-bad.rs:5:10
   |
LL | #[derive(IntoUnderlying)]
   |          ^^^^^^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[FIXME]: `IntoUnderlying` can only be derived for enums with an explicit integer representation
  --> $DIR/deriving-into-underlying-bad.rs:9:10
   |
LL | #[derive(IntoUnderlying)]
   |          ^^^^^^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[FIXME]: `IntoUnderlying` can only be derived for enums with an explicit integer representation
  --> $DIR/deriving-into-underlying-bad.rs:17:10
   |
LL | #[derive(IntoUnderlying)]
   |          ^^^^^^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 3 previous errors

